socket_linux.go

```diff
@@ -184,7 +184,7 @@ func SocketDiagTCPInfo(family uint8) ([]*InetDiagTCPInfoResp, error) {
 	req.AddData(&socketRequest{
 		Family:   family,
 		Protocol: unix.IPPROTO_TCP,
-		Ext:      INET_DIAG_INFO,
+		Ext:      (1 << (INET_DIAG_VEGASINFO - 1)) | (1 << (INET_DIAG_INFO - 1)),
```

Comment:
Is it ok for `SocketDiagTCPInfo()` to now return both _INET_DIAG_INFO_ and _INET_VEGAS_INFO_?
I see there is a bunch of DIAG_INFO options. Would the two above go hand in hand for TCP or could other ones also apply? Otherwise should we export a function that let the caller decide which INFO to pull, the Ext bitbask?

Reply:
It would be more flexible if the user could specify the options/bitmask.
However, specifying flags that are not parsed (yet) could then be interpreted as a bug.
Also dealing with bitmasks is not very straight forward for many. So I think this lib should definitely hide it somehow.

I think primary question is: Do we care if the routine responds with additional data that a user didn't ask for explicitly?
If yes I'd make that configurable via an Options struct/parameter.

Reply:
>Also dealing with bitmasks is not very straight forward for many. So I think this lib should definitely hide it somehow.

Yes, I see in the respective kernel code the bitmask creation is a bit quirky (`1<< (enum -1)`).
I was hoping in this library we could define those const values directly as bitmask, so then user can call our method with `const1 | const2 | ...` as they are already exported
```
const (
	INET_DIAG_NONE = 1
	INET_DIAG_MEMINFO = 2
        INET_DIAG_INFO = 4
...
```
This way we do not need to define any option setter function or option parsing logic.
WDYT?

Reply:
Honestly, I think when someone takes a look into TCP connection statistics they are usually happy to also get congestion avoidance aglorithm data like those for BBR. On the other hand I can't really think of anyone being unhappy about that.

So my conclusion is to just leave it as it is now as it is probably not getting any simpler.

Reply:
True, actually the function is for `TCPInfo`. It makes sense to return any diag info related to TCP.